### PR TITLE
Add CMS interaction to tests

### PR DESCRIPTION
**Pull request by andrewfrench**
## The Problem/Issue/Bug:
Our tests previously didn't interact with the CMS applications being deployed to confirm they actually worked.

## How this PR Solves The Problem:
This PR provides Interactor, an interface that provides methods to test interactivity of an app using the basic points outlined in #677. 

## Manual Testing Instructions:
The vast majority of changes here are updates to vendor/ in order to use [Surf](https://github.com/headzoo/surf), a programmatic browser for Go that handles interaction with a web page and is super cool.

## Automated Testing Overview:
Examples of use are found in start_test.go.

### CODE REVIEW

**Comment by andrewfrench** [comment]
Opening this PR to get some early opinions and test coverage. Very much a work in progress.

File: pkg/testinteraction/common.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
+package testinteraction
```
Comment (rfay) [inline pkg/testinteraction/common.go]: I would have expected to see this in testcommon, but don't have any objection if this is where you want it.

File: pkg/testinteraction/common.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	"regexp"
+
+	"github.com/drud/ddev/pkg/exec"
+	"github.com/headzoo/surf/browser"
+)
+
+// commonInteraction holds values that all interaction structs will need, allowing for common
+// methods to be defined.
+type commonInteraction struct {
+	adminUsername string
```
Comment (rfay) [inline pkg/testinteraction/common.go]: I'm not all the way through this yet, but is it important that it be an adminUsername or just a provided and valid user/pass?

**Comment by andrewfrench** [comment]
This is absolutely intended for use by ordinary tests, it's just packaged independently (which is definitely not required). The idea is that each app type will support the same general steps (configure, install, login, visit), but each will have their own workflow and criteria, so each should implement the specifics on its own.

I'm going to close this for now as full test coverage uncovered some issues that might take a good amount of work to resolve, but I expect this to resurface in one form or another sometime soon.

**Pull request closed without merging**